modules/openbsd/openbsd-grammar.ym

```diff
@@ -0,0 +1,119 @@
+
+/*
+ * Copyright (c) 2018 Balabit Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+%code top {
+#include "openbsd-parser.h"
+#include "openbsd-driver.h"
+
+}
+
+
+%code {
+
+#include "cfg-parser.h"
+#include "openbsd-grammar.h"
+#include "syslog-names.h"
+#include "messages.h"
+#include "plugin.h"
+#include "cfg-grammar.h"
+
+#include <string.h>
+
+}
+
+%name-prefix "openbsd_"
+
+/* this parameter is needed in order to instruct bison to use a complete
+ * argument list for yylex/yyerror */
+
+%lex-param {CfgLexer *lexer}
+%parse-param {CfgLexer *lexer}
+%parse-param {LogDriver **instance}
+%parse-param {gpointer arg}
+
+/* INCLUDE_DECLS */
+
+%type <ptr> source_openbsd
+%type <ptr> source_openbsd_params
+
+%token KW_OPENBSD
+%token KW_FREQUENCY
+%token KW_MESSAGE_TEMPLATE
+
+%%
+
+start
+  : LL_CONTEXT_SOURCE source_openbsd
+    {
+      YYACCEPT;
+    }
+  ;
+
+
+source_openbsd
+	: KW_OPENBSD '(' source_openbsd_params ')'	{ $$ = $3; }
+	;
+
+source_openbsd_params
+	: string
+	  {
+	    last_driver = *instance = openbsd_sd_new($1, configuration);
+	  }
+	  source_openbsd_options		{ $$ = last_driver; free($1); }
+	;
+
+source_openbsd_options
+	: source_openbsd_option source_openbsd_options
+	|
+	;
+
+source_openbsd_option
+	: 
+	;
+
+
+
+/*
+source_openbsd
+  : KW_OPENBSD '(' 
+    {
+      last_driver = *instance = openbsd_sd_new(configuration);
+      last_source_options = openbsd_sd_get_option(last_driver);
+    }
+     openbsd_options ')'
+  ;
+
+openbsd_options
+  : openbsd_option openbsd_options
+  |
+  ;
+
+openbsd_option
+  : KW_MESSAGE_TEMPLATE '(' template_content ')' { openbsd_option_set_template(last_source_options, $3);  }
```

Comment:
This source should not have `template` option at all, nor options.

Reply:
Removed